src/structures/tankWmech.jl

```diff
@@ -140,11 +144,278 @@ function tankWmech(gee::Float64, ρfuel::Float64,
+            end
+      end
+      kmax, imax = findmax(abs.(k))
+      ϕmax = ϕlist[imax]
+      return ϕmax, kmax
+end
+
+"""
+    stiffeners_bendingM_outer(θ1,θ2)
+This function can be used to calculate the bending moment distribution in a stiffener ring for an outer cryogenic tank.
+It applies Eqs. (7.13)-(7.15) in Barron (1985) to find the bending moment distribution. The function returns the
+maximum value of ``k = 2πM/(WR)`` on the ring's circumference.
+
+!!! details "🔃 Inputs and Outputs"
+    **Inputs:**
+    - `θ1::Float64`: angular position of bottom tank supports, measured from the bottom of the tank (rad).
+    - `θ2::Float64`: angular position of top tank supports, measured from the bottom of the tank (rad).
+
+    **Outputs:**
+    - `ϕmax::Float64`: angular position of maximum bending moment on ring circumference (rad).
+    - `kmax::Float64`: Maximum value of the ratio ``k = 2πM/(WR)`` on ring circumference.
+"""
+function stiffeners_bendingM_outer(θ1::Float64, θ2::Float64)
+      ϕlist = LinRange(0.0, π, 361)
+      k = zeros(length(ϕlist))
+
+      for (i,ϕ) in enumerate(ϕlist)
+            if 0 ≤ ϕ ≤ θ1
+                  k[i] = cos(ϕ)*(sin(θ2)^2 - sin(θ1)^2 ) + (cos(θ2) - cos(θ1) ) +
+                              - (π-θ2)*sin(θ2) +  (π-θ1)*sin(θ1)
+            elseif θ1 ≤ ϕ ≤ θ2
+                  k[i] = cos(ϕ)*(sin(θ2)^2 - sin(θ1)^2 ) + (cos(θ2) - cos(θ1) ) +
+                              - (π-θ2)*sin(θ2) +  π*sin(ϕ) -θ1*sin(θ1)
+            elseif θ2 ≤ ϕ ≤ π
+                  k[i] = cos(ϕ)*(sin(θ2)^2 - sin(θ1)^2 ) + (cos(θ2) - cos(θ1) )  +
+                              (θ2*sin(θ2) - θ1*sin(θ1)  )
+            end
+      end
+      kmax, imax = findmax(abs.(k))
+      ϕmax = ϕlist[imax]
+      return ϕmax, kmax
+end
+
+"""
+    optimize_outer_tank(fuse_tank, Winnertank, l_cyl, θ1, θ2)
+This function optimizes the number of intermediate stiffener rings to minimize the weight of an outer vessel.
+
+!!! details "🔃 Inputs and Outputs"
+    **Inputs:**
+    - `fuse_tank::Struct`: structure with tank parameters.
+    - `Winnertank::Float64`: weight of inner vessel and contents (N).
+    - `l_cyl::Float64`: length of cylindrical portion of outer vessel (m).
+
+    **Outputs:**
+    - `Ninterm::Float64`: optimum number of intermediate stiffener rings.
+"""
+function optimize_outer_tank(fuse_tank, Winnertank::Float64, l_cyl::Float64)
+
+      obj(x, grad) = size_outer_tank(fuse_tank, Winnertank, l_cyl, x[1])[1] #Minimize Wtank
+
+      initial_x = [fuse_tank.Ninterm]
+
+      #Set bounds
+      lower = [0.0]
+      upper = [50.0]
+
+      #Use NLopt.jl to minimize function 
+      opt = Opt(:LN_NELDERMEAD, length(initial_x))
+      opt.lower_bounds = lower
+      opt.upper_bounds = upper
+      opt.ftol_rel = 1e-9
+      opt.maxeval = 100  # Set the maximum number of function evaluations
+
+      opt.min_objective = obj
+
+      (minf,xopt,ret) = NLopt.optimize(opt, initial_x)
+      Ninterm = xopt[1]
```

Comment:
This will be a `float`! Shouldn't we take the `ceil`?
```suggestion
      Ninterm = ceil(Int, xopt[1])
```

Reply:
I decided to keep it a Float64 instead of ceiling it to an integer. The reasoning was that making it an integer would introduce discontinuities and I was concerned it would affect the sizing loop convergence.